android/library/src/main/java/io/logto/android/storage/CredentialStorage.kt

```diff
@@ -2,18 +2,18 @@ package io.logto.android.storage
 
 import android.content.Context
 import com.google.gson.Gson
+import io.logto.android.config.LogtoConfig
 import io.logto.android.constant.StorageKey
 import io.logto.android.model.Credential
 
 class CredentialStorage(
-    context: Context
+    context: Context,
+    logtoConfig: LogtoConfig,
 ) {
-    private val sharedPreferences by lazy {
-        context.getSharedPreferences(SHARED_PREFERENCE_NAME, Context.MODE_PRIVATE)
-    }
+    private val sharedPreferenceName = "$SHARED_PREFERENCE_NAME_PREFIX-${logtoConfig.clientId}"
```

Comment:
this should be more specific, add a linear issue to get persist key hash for config and think about how it works when user tries to create for a same config twice (throw an error maybe)

Reply:
TBD: 
[LOG-151](https://linear.app/silverhand/issue/LOG-151/enhance-process-correctly-when-use-the-same-logtoconfig-to-instantiate)
[LOG-150](https://linear.app/silverhand/issue/LOG-150/improvement-use-hash-to-make-logto-config-key-unique)